packages/klevu-ui/src/components/klevu-facet/klevu-facet.tsx

```diff
@@ -223,44 +235,73 @@ export class KlevuFacet {
           {this.labelOverride || this.option.label}
         </klevu-typography>
         <div class="options" slot="content">
-          {opts.map((o) => (
-            <div class="option">
-              {this.mode === "checkbox" ? (
-                <klevu-checkbox
-                  checked={o.selected}
-                  name={this.option!.key}
-                  onKlevuCheckboxChange={(event: CustomEvent<boolean>) => {
-                    this.manager.toggleOption(this.option!.key, o.value)
-                  }}
-                >
+          {this.useColorSwatch ? (
+            <div style={{ display: "flex", "flex-wrap": "wrap" }}>
+              {opts.map((o) => {
+                let swatch: KlevuSwatch = {
+                  name: o.name,
+                  color: o.value,
+                  selected: o.selected,
+                }
+                const override = this.colorSwatchOverrides.find((override) => override.name === o.name)
```

Comment:
In this code we have `opts.map` loop with `colorSwatchOverrides.find()`  inside of it that. This is run on every render and it can be quite unperformant if there are lots of options and overrides.

I suggest that we change the data model of `colorSwatchOverrides` to be object map instead of using `name` to find the override. 

If the Override type would be something like this: 
```ts
type KlevuColorSwatchOverride = {
 [key: string]: {
  color?: string
  imageUrl?: string
 }
}
```
 Then we wouldn't need to use `.find()` but `this.colorSwatchOverrides[o.name]`

Reply:
converted to suggested format